qiskit/transpiler/target.py

```diff
@@ -319,19 +326,33 @@ def add_instruction(self, instruction, properties=None, name=None):
                 documentation for the :class:`~qiskit.transpiler.Target` class).
         Raises:
             AttributeError: If gate is already in map
+            TranspilerError: If an operation class is passed in for ``instruction`` and no name
+                is specified.
         """
         if properties is None:
             properties = {None: None}
-        instruction_name = name or instruction.name
+        is_class = inspect.isclass(instruction)
+        if not is_class:
+            instruction_name = name or instruction.name
+        else:
+            # Invalid to have class input without a name with characters set "" is not a valid name
+            if not name:
+                raise TranspilerError(
+                    "A name must be specified when defining a supported global operation by class"
+                )
+            instruction_name = name
         if instruction_name in self._gate_map:
             raise AttributeError("Instruction %s is already in the target" % instruction_name)
         self._gate_name_map[instruction_name] = instruction
-        qargs_val = {}
-        for qarg in properties:
-            if qarg is not None:
-                self.num_qubits = max(self.num_qubits, max(qarg) + 1)
-            qargs_val[qarg] = properties[qarg]
-            self._qarg_gate_map[qarg].add(instruction_name)
+        if is_class:
+            qargs_val = {None: None}
+        else:
+            qargs_val = {}
+            for qarg in properties:
+                if qarg is not None:
+                    self.num_qubits = max(self.num_qubits, max(qarg) + 1)
+                qargs_val[qarg] = properties[qarg]
+                self._qarg_gate_map[qarg].add(instruction_name)
```

Comment:
Should we handle the case that a user passes a `properties` argument with a class?  I feel like `{None: None}` is the only thing that makes sense for control-flow, but I could potentially imagine `mcx` wanting to specify the allowed combinations of qubits directly _maybe_?

Happy just to make it a warning/error for this release, though, and think more about the interface later.

Reply:
yeah, I think we should error in that case, my feeling was if properties were passed in they should be using an instance of the gate. But we can discuss this for 0.23.0 if we need something different, an exception is easy to roll back if we want to allow it in the future.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/8852/commits/6ce7fe6375d34bf879ed9aac7bcdae2d3102179d